source/code/plugins/oms_common.rb

```diff
@@ -575,33 +691,23 @@ def get_os_version(conf_path = "/etc/opt/microsoft/scx/conf/scx-release")
         return @@OSVersion
       end
 
-      def get_hostname
-        return @@Hostname if !@@Hostname.nil?
+      def get_hostname(ignoreOldValue=false)
+        # Notes:  This implementation does not try to address multi-byte character aspects,
+        # but merely tries to address simple exigencies.  There may be multi-byte issues
+        # we will need to address in the future here.
+        if not is_hostname_compliant?(@@Hostname) or ignoreOldValue then
 
-        # Issue:
-        #   When omsagent runs inside a container, gethostname returns the hostname of the container (random name)
-        #   not the actual machine hostname.
-        #   One way to solve this problem is to set the container hostname same as machine name, but this is not
-        #   possible when host-machine is a private VM inside a cluster.
-        # Solution:
-        #   Share/mount ‘/etc/hostname’ as '/var/opt/microsoft/omsagent/state/containername' with container and
-        #   omsagent will read hostname from shared file.
+            look_in_designated_hostnamefile             if has_designated_hostnamefile?
 
-        begin
-          if File.exist?(@@HostnameFilePath) && File.readable?(@@HostnameFilePath)
-            @@Hostname = File.read(@@HostnameFilePath).strip
-            return @@Hostname
-          end
-        rescue => error
-          OMS::Log.warn_once("Unable to read the hostname from #{@@HostnameFilePath}: #{error}")
-        end
+            look_for_socket_class_host_address      unless is_hostname_compliant?(@@Hostname)
```

Comment:
style nit: use a single space between the method and conditional here and on line 700